arch/arm/core/cortex_m/mpu/arm_mpu.c

```diff
@@ -93,15 +94,33 @@ static inline int _get_region_attr_by_type(arm_mpu_region_attr_t *p_attr,
 	u32_t type, u32_t base, u32_t size)
 {
 	switch (type) {
+#ifdef CONFIG_MPU_REQUIRES_NON_OVERLAPPING_REGIONS
+#ifdef CONFIG_APP_SHARED_MEM
+	case APP_SHARED_MEM_BKGND_REGION:
+#endif
+	case KERNEL_BKGND_REGION:
+		_get_mpu_ram_region_attr(p_attr, P_RW_U_NA, base, size);
+		return 0;
+#endif /* CONFIG_MPU_REQUIRES_NON_OVERLAPPING_REGIONS */
 #ifdef CONFIG_USERSPACE
 	case THREAD_STACK_REGION:
 		_get_mpu_ram_region_attr(p_attr, P_RW_U_RW, base, size);
 		return 0;
+#ifdef CONFIG_MPU_REQUIRES_NON_OVERLAPPING_REGIONS
+	case KERNEL_BKGND_THREAD_STACK_REGION:
+		_get_mpu_ram_region_attr(p_attr, P_RW_U_NA, base, size);
+		return 0;
+#endif /* CONFIG_MPU_REQUIRES_NON_OVERLAPPING_REGIONS */
 #endif
 #ifdef CONFIG_MPU_STACK_GUARD
 	case THREAD_STACK_GUARD_REGION:
 		_get_mpu_ram_region_attr(p_attr, P_RO_U_NA, base, size);
 		return 0;
+#ifdef CONFIG_MPU_REQUIRES_NON_OVERLAPPING_REGIONS
+	case KERNEL_BKGND_STACK_GUARD_REGION:
+		_get_mpu_ram_region_attr(p_attr, P_RW_U_NA, base, size);
```

Comment:
does this need to be P_RO_U_NA?

Reply:
@agross-linaro, no, it does not; this is the region covering the entire Kernel SRAM (containing all thread stacks and global objects). It should be available for privileged RW.

Note that this region is supposed to have identical access permissions, as the "fixed" SRAM0, SRAM1, ... regions defined in the SOC-definitions of ARMv7-M SOCs

Reply:
ARMv8-M does not need these fixed regions at all, because it ensures that all SRAM is partitioned with non-overlapping regions (Kernel, Application Memory, Shared App Memory)